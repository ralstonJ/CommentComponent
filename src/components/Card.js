import React from "react";
import { MdExpandLess, MdExpandMore } from "react-icons/md";

const Card = props => {
  const {
    name,
    logo,
    type,
    time,
    message,
    nReply,
    upVote,
    downVote
  } = props.profile;
  return (
    <div className="card">
      <div>
        <img width={40} height={40} src={logo} className="card-profile-img" />
      </div>
      <div className="card-content">
        <div className="card-header">
          {name}
          {type === "Author" && <span className="card-type">Author</span>}{" "}
          {time} ago
        </div>
        <div className="card-message">{message}</div>
        <div className="card-footer">
          <button className="btn">REPLY</button>
          <button className="btn">
            {nReply} {nReply === 1 ? "REPLY" : "REPLIES"}
          </button>
          <button className="btn">
            <MdExpandLess className="vote-icon" />
            <span className="txt-vote">{upVote}</span>
          </button>
          <button className="btn">
            <MdExpandMore className="vote-icon" />
            <span className="txt-vote">{downVote}</span>
          </button>
        </div>
      </div>
    </div>
  );
};

export default Card;
