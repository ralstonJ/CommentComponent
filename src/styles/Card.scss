.card {
  display: flex;
  width: 500px;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
}

.card-profile-img {
  border-radius: 40px;
  background: black;
  margin: 10px;
}

.card-type {
  margin: 0px 10px;
  border-radius: 10px;
  background: black;
  padding: 0px 10px;
  color: white;
}

.card-content {
  margin: 10px;
}

.card-header {
  display: flex;
}

.card-message {
  text-align: left;
  margin: 10px 0px;
}

.card-footer {
  display: flex;
  font-size: 11px;
}

.btn {
  color: #808080;
  margin-right: 10px;
  cursor: pointer;
  border: none;
  padding: 0px;
  .material-icons.md-dark {
    color: #808080;
  }
}
.card:hover {
  .btn {
    color: #282828;
    .material-icons.md-dark {
      color: #282828;
    }
  }
}

.vote-icon {
  font-size: 22px;
  position: relative;
  top: 2px;
}

.txt-vote {
  position: relative;
  top: -5px;
}
