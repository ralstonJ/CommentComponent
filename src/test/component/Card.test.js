import React from "react";
import { shallow, configure, mount, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Card from "../../components/Card";
import logo from "../../logo.svg";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });
describe("Card", () => {
  it("should match the snapshot with all data passed in correctly", () => {
    const component = shallow(
      <Card
        profile={{
          logo,
          name: "Brad",
          type: "Author",
          time: "2 minutes",
          message:
            "So what the German automaker is likely to focus on today is the bigger picture. This will be the first time we see the Taycan free from any prototype bodywork.",
          nReply: 21,
          upVote: 123,
          downVote: 0
        }}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
