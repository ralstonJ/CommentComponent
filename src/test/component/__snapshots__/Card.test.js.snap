// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card"
>
  <div>
    <img
      className="card-profile-img"
      height={40}
      src="logo.svg"
      width={40}
    />
  </div>
  <div
    className="card-content"
  >
    <div
      className="card-header"
    >
      Brad
      <span
        className="card-type"
      >
        Author
      </span>
       
      2 minutes
       ago
    </div>
    <div
      className="card-message"
    >
      So what the German automaker is likely to focus on today is the bigger picture. This will be the first time we see the Taycan free from any prototype bodywork.
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn"
      >
        REPLY
      </button>
      <button
        className="btn"
      >
        21
         
        REPLIES
      </button>
      <button
        className="btn"
      >
        <MdExpandLess
          className="vote-icon"
        />
        <span
          className="txt-vote"
        >
          123
        </span>
      </button>
      <button
        className="btn"
      >
        <MdExpandMore
          className="vote-icon"
        />
        <span
          className="txt-vote"
        >
          0
        </span>
      </button>
    </div>
  </div>
</div>
`;
